lib/supavisor/db_handler.ex

```diff
@@ -114,11 +117,11 @@ defmodule Supavisor.DbHandler do
                 ]
 
                 bin = :pgo_protocol.encode_scram_response_message(sasl_initial_response)
-                :gen_tcp.send(data.socket, bin)
+                sock_send(data.sock, bin)
```

Comment:
consider checking return and gracefully handling errors. Or just crash if it wasn't successful.

```suggestion
                :ok = sock_send(data.sock, bin)
```